package/MDAnalysis/analysis/pca.py

```diff
@@ -0,0 +1,228 @@
+    variance : array (n_components, )
+        The raw variance explained by each eigenvector of the covariance
+        matrix.
+    explained_variance : array, (n_components, )
+        Percentage of variance explained by each of the selected components.
+        If a subset of components is not chosen then all components are stored
+        and the sum of explained variances is equal to 1.0.
+
+    Methods
+    -------
+    fit(traj=None, select='All', start=None, stop=None, step=None)
+        Find the principal components of selected atoms in a subset of the
+        trajectory.
+    transform(traj=None, n_components=None)
+        Take the original trajectory and project it onto the principal
+        components.
+    inverse_tranform(pca_space)
+        Take a pca_space and map it back onto the trajectory used to create it.
+    """
+
+    def __init__(self, u, select='all'):
+        """
+        Parameters
+        ----------
+        u : MDAnalysis Universe
+            The universe containing the trajectory frames for Principal
+            Component Analysis.
+        """
+        self._u = u
+        self._atoms = self._u.select_atoms(select)
+        self._n_atoms = self._atoms.n_atoms
+        self._calculated = False
+
+
+    def fit(self, n_components=None, start=None, stop=None,
+            step=None):
+        """ Use a subset of frames from the trajectory to generate the
+            principal components.
+
+        Parameters
+        ----------
+
+        n_components : int, optional
+            The number of principal components to be saved, default saves
+            all principal components, Default: -1
+        start : int, optional
+            First frame of trajectory to use for generation
+            of covariance matrix, Default: 0
+        stop : int, optional
+            Last frame of trajectory to use for generation
+            of covariance matrix, Default: -1
+        step : int, optional
+            Step between frames of trajectory to use for generation
+            of covariance matrix, Default: 1
+
+        Return
+        ------
+        cumulated_variance: array, (n_components, )
+            The amount of variance explained by the nth component and the n-1
+            components preceding it.
+        p_components: array, (n_components, n_atoms * 3)
+
+        """
+        self._setup_frames(self._u.trajectory, start, stop, step)
+        self.n_components = n_components
+        self.run()
+        self._calculated = True
+
+        if n_components is None:
+            return self.cumulated_variance, self.p_components
+        else:
+            return (self.cumulated_variance[:n_components],
+                    self.p_components[:n_components])
+
+    def _prepare(self):
+        self._xyz = np.zeros((self.n_frames, self._n_atoms, 3))
+
+
+    def _single_frame(self):
+        self._xyz[self._frame_index] = self._atoms.positions.copy()
```

Comment:
What I'm saying is that the sum goes over the frames. So you have to slowly build up the sum for each entry of the matrix as you iterate over the frame. Something like this should do the trick.

```python
def _prelude(self):
    n_dim = self.atoms.n_atom * 3
    self.cov = np.zeros(n_dim, n_dim)

def _single_frame(self):
     x = self.atoms.positions.flat  # maybe you need to use ravel as this only iterates!
     self.cov += np.dot(x[:, np.newaxis], x[:, np.newaxis].T)

def _conclude(self):
    self.cov /= self.n_frames - 1
```

Please check the dot product. To get more information about the correct normalization have a look at the [np.cov docs](http://docs.scipy.org/doc/numpy/reference/generated/numpy.cov.html)